psa-crypto/src/operations/asym_signature.rs

```diff
@@ -2,13 +2,57 @@
 // SPDX-License-Identifier: Apache-2.0
 
 //! # Asymmetric Signature operations
+//!
+//! See the PSA Crypto API for the format of the different parameters used in this module.
 
 use crate::initialized;
 use crate::types::algorithm::AsymmetricSignature;
 use crate::types::key::Id;
 use crate::types::status::{Result, Status};
 
-/// Sign a hash
+/// Sign an already-calculated hash with a private key
+///
+/// The signature is written in `signature`. The function returns the number of bytes written.
+///
+/// # Example
+///
+/// ```
+/// # use psa_crypto::operations::key_management::generate;
+/// # use psa_crypto::operations::asym_signature::sign_hash;
+/// # use psa_crypto::types::key::{Attributes, Type, Lifetime, Policy, UsageFlags};
+/// # use psa_crypto::types::algorithm::{AsymmetricSignature, Hash};
+/// # let mut attributes = Attributes {
+/// #     key_type: Type::RsaKeyPair,
+/// #     bits: 1024,
+/// #     lifetime: Lifetime::Volatile,
+/// #     policy: Policy {
+/// #         usage_flags: UsageFlags {
+/// #             sign_hash: true,
+/// #             sign_message: true,
+/// #             verify_hash: true,
+/// #             verify_message: true,
+/// #             ..Default::default()
+/// #         },
+/// #         permitted_algorithms: AsymmetricSignature::RsaPkcs1v15Sign {
+/// #             hash_alg: Hash::Sha256.into(),
```

Comment:
Why do this line and the following line contain `#`?

Reply:
It is made to hide portions of the example so it only shows relevent parts. See [here](https://doc.rust-lang.org/rustdoc/documentation-tests.html#hiding-portions-of-the-example) for details.